sycl/include/CL/sycl/handler.hpp

```diff
@@ -505,7 +505,7 @@ class __SYCL_EXPORT handler {
             access::placeholder IsPH>
   detail::enable_if_t<Dim == 0 && Mode == access::mode::atomic, T>
   readFromFirstAccElement(accessor<T, Dim, Mode, Target, IsPH> Src) const {
-    atomic<T, access::address_space::global_space> AtomicSrc = Src;
+    atomic<T, access::address_space::device_space> AtomicSrc = Src;
```

Comment:
Will it work for host device?

Reply:
It shan't be matter for host, since for x86 target clang will generate addr space 0 for any pointer.